var/spack/repos/builtin/packages/blast-plus/package.py

```diff
@@ -0,0 +1,98 @@
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+from spack import *
+
+
+class BlastPlus(Package):
+    """BLAST (Basic Local Alignment Search Tool) finds regions of similarity
+    between biological sequences. The program compares nucleotide or protein sequences
+    to sequence databases and calculates the statistical significance"""
+
+    homepage = "http://ncbi.nlm.nih.gov/Blast.cgi"
+
+    variant("static", default=True, description="build static libraries and executables")
+    variant("freetype", default=False, description="build with freetype")
+    variant("gnutls", default=False, description="enable gnutls")
+    variant("hdf5", default=False, description="enable hdf5")
+    variant("jpeg", default=True, description="enable jpeg")
+    variant("libpng", default=True, description="enable libpng")
+    variant("pcre", default=True, description="enable pcre")
+    variant("python", default=True, description="build with python support")
+    variant("debug", default=True, description="Enable debug")
+
+    version('2.6.0', 'c8ce8055b10c4d774d995f88c7cc6225', url="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/ncbi-blast-2.6.0+-src.tar.gz")
+    version('2.5.0', '54ad4f2ea15715487b3de712a9d27be8',  url="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.5.0/ncbi-blast-2.5.0+-src.tar.gz")
+
+    depends_on("freetype", when="+freetype")
+    depends_on("gnutls", when="+gnutls")
+    depends_on("python", when="+python")
+    depends_on("hdf5", when="+hdf5")
+    depends_on("jpeg", when="+jpeg")
+    depends_on("pcre", when="+pcre")
+    depends_on("libpng", when="+libpng")
+
+
+    # Run patch to prevent make install from failing. 
+    # Patch and problem were solved in this thread:
+    # https://github.com/Homebrew/homebrew-science/pull/4740
+    patch('blast-plus-make-fix.patch', when="@2.5.0:")
+
+    def install(self, spec, prefix):
+        config_args = ["--prefix=" + prefix,
+                       "--with-bin-release",
+                       "--with-mt",
+                       "--with-64",
+                       "--without-debug",
+                       "--with-optimization"]
+
+        if "+static" in spec:
+            config_args.append("--with-static")
+            config_args.append("LDFLAGS=-static")
+            if not spec.satisfies("platform=linux"):
+                config_args.append("--with-static-exe")
+        else:
+            config_args.extend(["--without-static", "--without-static-exe"])
+
+        if "+freetype" in spec:
+            config_args.append("--with-freetype=%s" % self.spec['freetype'].prefix)
```

Comment:
For all of these options, is there a `--without-` equivalent? If so, you'll want to add it. A lot of times if you don't specify, it will activate the flag if it finds the library and deactivate it if it doesn't. Meaning that `blast-plus +freetype` and `blast-plus -freetype` could both have freetype support if the library is present on your system.